proposals/002-text-utf-default.md

```diff
@@ -0,0 +1,156 @@
+This proposal is for the migration of the `text` package from its
+default UTF-16 encoding to UTF-8. The lack of UTF-8 as a default in the
+`text` package is a pain point raised by the Haskell Community and
+industry partners for many years. The Haskell Foundation is uniquely
+positioned to effect a change like this, granted there is an appetite
+for breakage, and that such a project is well-socialized and planned.
+During the meetings of the Haskell Foundation Tech Agenda Track, we
+identified the pros and cons of such a migration against historical
+attempts, and agreed that it was within our power to do.
+
+Further, we solicited both community and industry feedback to gauge the
+appetite for breakage, and what stakeholders would be affected by the
+changes. Andrew Lelechenko has offered to lead the implementation of
+this project.
+
+# Motivation
+
+-   UTF-16 by default requires that all Text values pay a premium for serialization. Arguably, the performance impact of Text is flipped
+    upside-down: most text is UTF-8, and Haskell devs pay an undue cost when working with the wrong default.
+
+-   UTF-8 is the industry standard and by far the most common text encoding, with roughly 97% of web pages existing in UTF-8. The
+    existing UTF-16 default imposes an additional hurdle to working with the vast majority of web content on earth.
+
+-   Many systems in Haskell are UTF-8 by default (e.g. Haddock)
+
+# Goals
+
+-   Solicit feedback from community members and industry to gauge appetite for such a change.
+
+-   Provide an implementation, migration, and delivery plan for changing the default encoding of `text` from UTF-16 to UTF-8.
+
+-   Ensure stakeholders (e.g. GHC, Cabal, Stack, boot libs) have ample time to migrate and address any bugs.
+
+-   Implementation should not significantly alter the performance characteristics of the base `text` library within some tolerance
```

Comment:
What tolerance threshold? 10x is "some" threshold, but probably not the one you want. I wouldn't see a violation of this threshold as a standalone reason to declare failure, but if we say the new implementation should be, say, within 5% of the performance of the old one, and it's an 8% regression, it would mean at least advertising this downside and seeking more community input.

Reply:
I think this is largely about managing expectations, because eg making sure changes in eg code size dont create differences in how inlining and fusion operate on otherwise observationaly equal programs is quite reasonably out of scope for this project (text is very dependent on good quality fusion optimization for good perf by most users, but making fusion rules more robust is out of scope for this work)

Reply:
All I'm saying is that we should name a specific goal threshold for how much regression we're willing to tolerate before we take a step back and wonder whether this was a good idea.

Reply:
ok, "commiting to a decision threshold where we should look at what is failing and why to learn from and fix" sounds like a good idea

Reply:
I don't think you should name a specific threshold. Rather, once the implementation is underway, and a sense of the tradeoffs is more apparent, _then_ a decision should be made, based on the facts at hand, not an arbitrary before-hand choice.

Reply:
I agree with @gbaz here. This is more of an implementation detail that I would need @Bodigrim to weigh in on once he can make a reasonable estimate.